src/DynamoCore/Configuration/PathManager.cs

```diff
@@ -131,7 +163,14 @@ public string CommonDataDirectory
 
         public string DefaultUserDefinitions
         {
-            get { return TransformPath(RootDirectories.First(), DefinitionsDirectoryName); }
+            get 
+            {
+                if (Preferences is PreferenceSettings preferences)
+                {
+                    return TransformPath(preferences.SelectedPackagePathForInstall, DefinitionsDirectoryName);
+                }
+                return TransformPath(RootDirectories.First(), DefinitionsDirectoryName); 
+            }
         }
 
         public IEnumerable<string> DefinitionDirectories
```

Comment:
I'm not sure what this should be? Currently it includes std lib directory and program data paths, which seems wrong.

Reply:
What are re using it for?

Reply:
Custom nodes.